drivers/flash/Kconfig.stm32_qspi

```diff
@@ -18,5 +18,7 @@ config FLASH_STM32_QSPI
 	select FLASH_HAS_PAGE_LAYOUT
 	select DMA if $(DT_STM32_QUADSPI_HAS_DMA)
 	select USE_STM32_HAL_DMA if $(DT_STM32_QUADSPI_HAS_DMA)
+	select USE_STM32_HAL_MDMA if !$(DT_STM32_QUADSPI_HAS_DMA)
```

Comment:
The proposed logic is not correct.
User always has the possibility of not using DMA wth QSPI, even if supported on the series in use.
With this change, MDMA is systematically enabled in this case.

Reply:
The MDMA is always required by the STM32 QSPI HAL driver also without DMA support. As a workaround to void compiler errors I enabled the MDMA HAL driver if DMA support is disabled.

Reply:
This maybe the case on H7, but not on all series.
You can have a try by doing the folloing change in disco_l475_iot1.dts:
```
 &quadspi {
 	pinctrl-0 = <&quadspi_clk_pe10 &quadspi_ncs_pe11
 		     &quadspi_bk1_io0_pe12 &quadspi_bk1_io1_pe13
 		     &quadspi_bk1_io2_pe14 &quadspi_bk1_io3_pe15>;
-	dmas = <&dma1 5 5 0x0000 0x03>;
-	dma-names = "tx_rx";
+	//dmas = <&dma1 5 5 0x0000 0x03>;
+	//dma-names = "tx_rx";
 
```
Then run:
```
west build -b  disco_l475_iot1 samples/drivers/flash_shell
```